test/python/memoize.py

```diff
@@ -0,0 +1,53 @@
+import inspect
+# Private import to format error messages.
+# if this changes, no big deal, just remove
+from _pytest.fixtures import fail_fixturefunc
+
+
+# Memoizes the value.
+# Works with generator or function.
+# has same restrictions on generators as pytest - can only yield once
+
+# I think this is the behavior you want, start up, value, teardown happens once in expected scope.
+# The value is memoized.
+def memoize(func):
+    cache = {}
+
+    def memoized_func(*args):
+        if 'value' not in cache:
+            value = func(*args)
+            if inspect.isgeneratorfunction(func):
+                cache['generator'] = value
+                value = next(value)
+            cache['value'] = value
+
+        return cache['value']
+
+    def generator_after():
+        if 'generator' in cache:
+            try:
+                next(cache['generator'])
+            except StopIteration:
+                cache.pop('generator')
+            else:
+                fail_fixturefunc(func, "yield_fixture function has more than one 'yield'")
+
+    arg_string = ",".join(inspect.getargspec(func)[0])
+
+    # We create a wrapper method to be able to bind the generated method (which mirrors the fixture method exactly)
+    # to the memoized_func and after handler
+    code = '''
```

Comment:
@rach for my own python betterment - is there a way to do this which doesn't use a string eval?